portfolio/src/main/webapp/request.js

```diff
@@ -64,9 +65,46 @@ function onClickStartScheduling() {
  * Updates the UI to show the results of a query.
  */
 function handleScheduledTaskArray(scheduledTaskArray) {
-  const resultElement = document.getElementById('scheduled-task-list');
+  const resultElement = document.getElementById('schedule-result-list');
   resultElement.innerHTML = '';
-  scheduledTaskArray.forEach(addScheduledTaskToDom, resultElement);
+  scheduledTaskArray.forEach(addScheduledTaskToDom);
+}
+
+/**
+ * Returns a JSON of the latest scheduled tasks pulled from the DOM.
+ */
+function collectAllScheduledTasks() {
+  const allScheduledTasks = new Array();
+  // Get scheduled task list element to run through cards collecting data.
+  const scheduledTaskList = document.getElementById('schedule-result-list');
+
+  scheduledTaskList.childNodes.forEach((scheduledTaskCard) => {
+    // The cardBody of the scheduledTaskCard
+    const cardBody = scheduledTaskCard.childNodes[0];
+
+    const scheduledTaskName = cardBody.childNodes[0].innerText;
```

Comment:
I would suggest that you store the data in some HTML element attribute, rather than inner text.

For example, the scheduled task name HTML element can have a data attribute, like,

data-task-name="..."

Then you don't have to do this substring operation. See the pages below for more info.

https://www.w3schools.com/tags/att_data-.asp
https://api.jquery.com/data/

Reply:
Done